code/game/objects/random/random.dm

```diff
@@ -9,9 +9,13 @@
 // creates a new object and deletes itself
 /obj/random/New()
 	..()
-	if (!prob(spawn_nothing_percentage))
-		spawn_item()
-	qdel(src)
+	spawn()
+		if(istype(src.loc, /obj/structure/loot_pile)) //Spawning from a lootpile is weird, need to wait until we're out of it to do our work.
+			while(istype(src.loc, /obj/structure/loot_pile))
```

Comment:
This spawn() in new, and the while loop that sleeps is not so great.  However we don't use /obj/structure/loot_pile so its not urgent to fix.